[receiver/jmxreceiver] add temp_folder to jmx receiver config (#17884)
open-telemetry/opentelemetry-collector-contrib: receiver/jmxreceiver/README.md

Corresponds to the `org.slf4j.simpleLogger.defaultLogLevel` property.

### temp_dir (default: operating system temporary directory)

> The location of the directory where the receiver will store configuration used by the Java process. The configuration is typically stored as a properties file, as a temporary file that is deleted on exit of the collector.

Reviewer: ```suggestion
The location of the directory where the receiver will store configuration used by the Java process. The configuration is typically stored as a properties file, which is deleted on exit of the collector.
```
